### PR TITLE
checksums.py: use importlib_resources

### LINKED ISSUE #2561: [Cleanup] Rewrite checksums.py to use `importlib_resources`
Currently, https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/download/checksums.py read checksums using relative dir: https://github.com/tensorflow/datasets/blob/0c308e4b7be9d0c834d0021c6cf6566f6cd57c00/tensorflow_datasets/core/download/checksums.py#L28

This is hacky when more modern https://docs.python.org/3/library/importlib.html#importlib.resources.files exists with a Pathlib-like interface.

```py
path = importlib_resources.files('tensorflow_datasets') / 'url_checksums'
files = [p for p in path.listdir() if str(p).endswith('txt')]
```

It would make TFDS work even when Python is executed inside compressed `.zip` archive.

### PR DESCRIPTION
**Pull request by iamsdas**
Fix #2561

### CODE REVIEW

**Comment by iamsdas** [comment]
fixed conflicts with master branch

File: tensorflow_datasets/core/download/checksums.py [hunk 1]
```diff
@@ -19,17 +19,17 @@
...
 
 from tensorflow_datasets.core import utils
 
 
-_ROOT_DIR = os.path.normpath(os.path.join(os.path.dirname(__file__), '../..'))
-
 _CHECKSUM_DIRS = [
-    os.path.join(_ROOT_DIR, 'url_checksums'),
+  os.path.normpath(
+    importlib_resources.files('tensorflow_datasets') / 'url_checksums'),
```
Comment (Conchylicultor) [inline tensorflow_datasets/core/download/checksums.py:32]: You could use: `utils.resources_path('tensorflow_datasets')` instead which is a wrapper of `importlib_resources.files`.

**Review by Conchylicultor — CHANGES_REQUESTED** [review]
Thank you for your contribution. Could you update the other methods to use pathlib-like methods like `path.read_text()` ?

File: tensorflow_datasets/core/download/checksums.py (hunk 1 above)
Comment (iamsdas) [reply tensorflow_datasets/core/download/checksums.py]: done

File: tensorflow_datasets/core/download/checksums.py [hunk 2]
```diff
@@ -19,16 +19,15 @@
...
```
[the remark this answers is not in the document]
Comment (iamsdas) [reply tensorflow_datasets/core/download/checksums.py]: added required changes, could you check it out?

**Review by Conchylicultor — CHANGES_REQUESTED** [review]
Could you run the tests and make sure they are passing ?

```sh
pytest tensorflow_datasets/core/download/
pytest tensorflow_datasets/core/
```

**Comment by iamsdas** [comment]
Fixed a typo `utils.resources_path -> utils.resource_path`. Now it passes all the tests.

**Pull request merged**